ts/src/header-validator/source.test.ts

```diff
@@ -1927,6 +1930,223 @@ const testCases: TestCase[] = [
       },
     ],
   },
+
+  // Attribution Scope
+  {
+    name: 'attribution-scope-limit-negative',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "attribution_scope_limit": -1
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must fit in an unsigned 32-bit integer'
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-zero',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "attribution_scope_limit": 0
+    }`,
+    expectedErrors: [
+      {
+        path: [],
+        msg: 'attribution scope limit should be positive.'
+      },
+    ],
+  },
+  {
+    name: 'missing-attribution-scope-limit-attribution-scopes',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "attribution_scopes": ["1", "2"]
+    }`,
+    expectedErrors: [
+      {
+        path: [],
+        msg: 'attribution scope limit should be set if attribution scopes or max event states are set'
+      },
+    ],
+  },
+  {
+    name: 'missing-attribution-scope-limit-max-event-states',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "max_event_states": 3 
+    }`,
+    expectedErrors: [
+      {
+        path: [],
+        msg: 'attribution scope limit should be set if attribution scopes or max event states are set'
+      },
+    ],
+  },
+  {
+    name: 'max-event-states-negative',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "attribution_scope_limit": 1,
+      "max_event_states": -1
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'must fit in an unsigned 32-bit integer'
+      },
+    ],
+  },
+  {
+    name: 'max-event-states-zero',
```

Comment:
Please also add test cases for non-integer values and for exceeding `MAX_UINT32`.

Reply:
Done.